app/namespace-watcher.go

```diff
@@ -45,77 +70,149 @@ func createLimitRange(clientset *kubernetes.Clientset, namespaceName string) err
 				{
 					Type: corev1.LimitTypePod,
 					Max: corev1.ResourceList{
-						corev1.ResourceCPU:              resource.MustParse(cpuLimitMax),
-						corev1.ResourceMemory:           resource.MustParse(memoryLimitMax),
-						corev1.ResourceEphemeralStorage: resource.MustParse(ephemeralStorageLimitMax),
+						corev1.ResourceCPU:              cpuLimitMax,
+						corev1.ResourceMemory:           memLimtMax,
+						corev1.ResourceEphemeralStorage: ephemeralStorageMax,
 					},
 					Min: corev1.ResourceList{
-						corev1.ResourceCPU:              resource.MustParse(cpuLimitMin),
-						corev1.ResourceMemory:           resource.MustParse(memoryLimitMin),
-						corev1.ResourceEphemeralStorage: resource.MustParse(ephemeralStorageLimitMin),
+						corev1.ResourceCPU:              cpuLimitMin,
+						corev1.ResourceMemory:           memLimtMin,
+						corev1.ResourceEphemeralStorage: ephemeralStorageMin,
 					},
 				},
 			},
 		},
 	}
 
-	_, err := clientset.CoreV1().LimitRanges(namespaceName).Create(context.Background(), limitRange, metav1.CreateOptions{})
+	// Get the existing LimitRange
+	existingLimitRange, err := clientset.CoreV1().LimitRanges(namespaceName).Get(context.Background(), limitRange.ObjectMeta.Name, metav1.GetOptions{})
 	if err != nil {
-		return err
+		if errors.IsNotFound(err) {
+
+			//Create LimitRange if it doesn't exist
+			_, err = clientset.CoreV1().LimitRanges(namespaceName).Create(context.Background(), limitRange, metav1.CreateOptions{})
+			if err != nil {
+				return err
+			}
+			logrus.Warn("Created LimitRange for namespace: ", namespaceName)
+		} else {
+			return err
+
+		}
+	} else {
+		// Update the existing LimitRange
+		existingLimitRange.Spec = limitRange.Spec
+		_, err = clientset.CoreV1().LimitRanges(namespaceName).Update(context.Background(), existingLimitRange, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+		logrus.Warn("Updated LimitRange for namespace: ", namespaceName)
 	}
 
-	fmt.Printf("Created LimitRange for namespace %s\n", namespaceName)
 	return nil
 }
 
+func lookupEnvOrEmpty(key string) string {
+	value, _ := os.LookupEnv(key)
+
+	return value
+}
+
 func main() {
 
+	logrus.Info("Namespace-Watcher version ", version)
+
+	//Evaluate if environment variables are not set or set to ""
+	setLimits.CpuLimitMin = lookupEnvOrEmpty("CPU_LIMIT_MIN")
+	if setLimits.CpuLimitMin == "" || len(strings.TrimSpace(setLimits.CpuLimitMin)) == 0 {
+		logrus.Fatal("Environment variable CPU_LIMIT_MIN is not set or empty: ", setLimits.CpuLimitMin)
```

Comment:
Some considerations about these conditional checks:
* Do not use `Fatal`, it causes the program to exit immediately
* Given you have a default in place, it's not even worth to mention that to the user. If you really want to, use the `Trace` or `Debug` level
* I don't think you have to do all this "massaging" of the user input
* There's a lot of repetition across these `if` statements. Maybe you can create a function like `parseValueOrDefault(envKey string, defaultValue resource.Quantity) (resource.Quantity, error)`

Reply:
I'm not putting a default in place.
I thought of Fatal because the watcher should fail/stop if the values can't be used?

Reply:
I see, I got confused. I though you had a default in place. It's up to you to decide what to do when one of these settings is missing.

Fine with me if you want to exit with an error, hence `Fatal` :smile: